src/python/module/smartredis/util.py

```diff
@@ -111,3 +114,93 @@ def typecheck(arg, name, _type):
     """
     if not isinstance(arg, _type):
         raise TypeError(f"Argument {name} is of type {type(arg)}, not {_type}")
+
+
+
+class DatasetConverter:   
+    
+    def add_metadata_for_xarray(ds,data=None,dims=None,coords=None,attrs=None): #time=None,time_variable=None):
```

Comment:
Also, I do suggest making the argument names  to be `data_names`, `dim_names`, etc. Otherwise I would assume that you want me to put in the actual values and not what field should be referenced in the dataset